src/Compilers/CSharp/Test/Semantic/SourceGeneration/SyntaxAwareGeneratorTests.cs

```diff
@@ -1960,6 +1960,37 @@ class C { }
             Assert.True(generatorCancelled);
         }
 
+        [Fact]
+        public void Syntax_Receiver_Cancellation_During_Visit()
+        {
+            var source = @"
+class C 
+{
+    int Property { get; set; }
+
+    void Function()
+    {
+        var x = 5;
+        x += 4;
+    }
+}
+";
+            var parseOptions = TestOptions.Regular;
+            Compilation compilation = CreateCompilation(source, options: TestOptions.DebugDll, parseOptions: parseOptions);
+            compilation.VerifyDiagnostics();
+
+            Assert.Single(compilation.SyntaxTrees);
+
+            var cts = new CancellationTokenSource();
+            var testGenerator = new CallbackGenerator(
+                onInit: (i) => i.RegisterForSyntaxNotifications(() => new TestSyntaxReceiver(tag: 0, callback: (a) => { if (a is AssignmentExpressionSyntax){ cts.Cancel(); cts.Token.ThrowIfCancellationRequested(); } })),
```

Comment:
This cancellation step is capturing the `CancellationTokenSource` from the outer context and using it to both cancel and detect cancellation. Is there no way for a receiver to grab the `CancellationToken` passed into `RunGeneratorsAndUpdateCompilation`? If so then I would expect the `ThrowIfCancellationRequested` call to be on that token (make sure we thread it through). 

If not how are syntax receivers expected to cancel?

Reply:
SyntaxReceivers do not have access to any cancellation tokens. All cancellation exceptions thrown by a syntax receiver are errors.

Reply:
I'm unsure what we are testing then here. If user code doesn't have access to the token then they can't exercise this code path. If our internal walk of the syntax throws based on the cancellation token passed to the generator then we shouldn't need a throw here.

Reply:
I'll adjust it to just throw a new cancellation exception to indicate its not expected to be coming from the driver. Honestly I think the set up is mostly just a copy paste from the execute test.